app/actors/hyrax/actors/asset_actor.rb

```diff
@@ -235,7 +235,9 @@ def set_validation_status(env)
           current_children_count = env.curation_concern.all_members.reject { |child| child.is_a?(Contribution) }.size
           intended_children_count = env.curation_concern.intended_children_count.to_i
 
-          if current_children_count < intended_children_count
+          if env.curation_concern.intended_children_count.nil? && env.curation_concern.validation_status_for_aapb.empty?
```

Comment:
```suggestion
          if env.curation_concern.intended_children_count.nil? && env.curation_concern.validation_status_for_aapb.blank?
```

I _think_ `#empty?` here might throw an error. I think there's a chance records created before the `:validation_status_for_aapb` property was introduced will return `nil` instead of `[]`. 

Even if that's not the case, I tend to almost always use `#blank?` since it covers more cases: 

```ruby
irb(main):001:0> nil.empty?
Traceback (most recent call last):
        1: from (irb):1
NoMethodError (undefined method `empty?' for nil:NilClass)

irb(main):002:0> nil.blank?
=> true
```

I'm not really worried about `intended_children_count.nil?` since `:intended_children_count` is singular and thus should be `nil` if it's unset (though I'd personally probably still use `#blank?` 😅)

Reply:
updated